packages/experimental/AppearanceAdditions/src/appearanceAdditions.ios.ts

```diff
@@ -4,37 +4,42 @@ import { memoize } from '@fluentui-react-native/framework';
 
 import NativeAppearanceAdditions from './NativeAppearanceAdditions';
 import type { AppearanceAdditions, SizeClass, UserInterfaceLevel, AccessibilityContrastOption } from './NativeAppearanceAdditions.types';
-import { HorizontalSizeClassKey, UserInterfaceLevelKey, AccessibilityContrastOptionKey } from './NativeAppearanceAdditions.types';
+import {
+  HorizontalSizeClassKey,
+  UserInterfaceLevelKey,
+  AccessibilityContrastOptionKey,
+  RootTagKey,
+} from './NativeAppearanceAdditions.types';
 
 class AppearanceAdditionsImpl implements AppearanceAdditions {
-  _horizontalSizeClass: SizeClass;
-  _userInterfaceLevel: UserInterfaceLevel;
-  _accessibilityContrastOption: AccessibilityContrastOption;
-
-  get horizontalSizeClass(): SizeClass {
-    return this._horizontalSizeClass;
-  }
-
-  get userInterfaceLevel(): UserInterfaceLevel {
-    return this._userInterfaceLevel;
-  }
-
-  get accessibilityContrastOption(): AccessibilityContrastOption {
-    return this._accessibilityContrastOption;
-  }
+  _rootTagHorizontalSizeClassMap: { [key: number]: SizeClass } = {};
+  _rootTagUserInterfaceLevelMap: { [key: number]: UserInterfaceLevel } = {};
+  _rootTagAccessibilityContrastOptionMap: { [key: number]: AccessibilityContrastOption } = {};
 
   constructor() {
-    this._horizontalSizeClass = NativeAppearanceAdditions.horizontalSizeClass();
-    this._userInterfaceLevel = NativeAppearanceAdditions.userInterfaceLevel();
-    this._accessibilityContrastOption = NativeAppearanceAdditions.accessibilityContrastOption();
+    this._rootTagHorizontalSizeClassMap = NativeAppearanceAdditions.horizontalSizeClass();
+    this._rootTagUserInterfaceLevelMap = NativeAppearanceAdditions.userInterfaceLevel();
+    this._rootTagAccessibilityContrastOptionMap = NativeAppearanceAdditions.accessibilityContrastOption();
 
     const eventEmitter = new NativeEventEmitter(NativeAppearanceAdditions as any);
     eventEmitter.addListener('appearanceChanged', (newValue) => {
-      this._horizontalSizeClass = newValue[HorizontalSizeClassKey];
-      this._userInterfaceLevel = newValue[UserInterfaceLevelKey];
-      this._accessibilityContrastOption = newValue[AccessibilityContrastOptionKey];
+      const rootTag = newValue[RootTagKey];
+
+      this._rootTagHorizontalSizeClassMap[rootTag] = newValue[HorizontalSizeClassKey];
+      this._rootTagUserInterfaceLevelMap[rootTag] = newValue[UserInterfaceLevelKey];
+      this._rootTagAccessibilityContrastOptionMap[rootTag] = newValue[AccessibilityContrastOptionKey];
     });
   }
+
+  horizontalSizeClassForRootTag(rootTag: number): SizeClass {
```

Comment:
Will these functions ever be called before the `eventEmitter` above has been invoked? If so, what are they returning?

Reply:
Hmm I don't think that can happen since those functions need to be called on an instance of AppearanceAdditions- so the constructor which initializes the eventEmitter would always be called first. Doing a very brief check, this seems to be the case. Were you thinking of a specific scenario?